miden-tx/src/host/mod.rs

```diff
@@ -394,20 +387,40 @@ impl<A: AdviceProvider, T: TransactionAuthenticator> Host for TransactionHost<A,
         }
 
         match event {
-            TransactionEvent::AccountVaultAddAsset => self.on_account_vault_add_asset(process),
-            TransactionEvent::AccountVaultRemoveAsset => {
-                self.on_account_vault_remove_asset(process)
+            TransactionEvent::AccountVaultBeforeAddAsset => Ok(()),
+            TransactionEvent::AccountVaultAfterAddAsset => {
+                self.on_account_vault_after_add_asset(process)
+            },
+
+            TransactionEvent::AccountVaultBeforeRemoveAsset => Ok(()),
+            TransactionEvent::AccountVaultAfterRemoveAsset => {
+                self.on_account_vault_after_remove_asset(process)
+            },
+
+            TransactionEvent::AccountStorageBeforeSetItem => Ok(()),
+            TransactionEvent::AccountStorageAfterSetItem => {
+                self.on_account_storage_after_set_item(process)
+            },
+
+            TransactionEvent::AccountStorageBeforeSetMapItem => Ok(()),
+            TransactionEvent::AccountStorageAfterSetMapItem => {
+                self.on_account_storage_after_set_map_item(process)
+            },
+
+            TransactionEvent::AccountBeforeIncrementNonce => {
+                self.on_account_before_increment_nonce(process)
             },
-            TransactionEvent::AccountStorageSetItem => self.on_account_storage_set_item(process),
-            TransactionEvent::AccountIncrementNonce => self.on_account_increment_nonce(process),
+            TransactionEvent::AccountAfterIncrementNonce => Ok(()),
+
             TransactionEvent::AccountPushProcedureIndex => {
                 self.on_account_push_procedure_index(process)
             },
-            TransactionEvent::NoteCreated => self.on_note_created(process),
-            TransactionEvent::AccountStorageSetMapItem => {
-                self.on_account_storage_set_map_item(process)
-            },
-            TransactionEvent::NoteAddAsset => self.on_note_add_asset(process),
+
+            TransactionEvent::BeforeNoteCreated => Ok(()),
+            TransactionEvent::AfterNoteCreated => self.on_note_after_created(process),
```

Comment:
Actually, looking at function names here, I think `NOTE_BEFORE_CREATED` and `NOTE_AFTER_CREATED` are fine.